workers/generic-worker/main.go

```diff
@@ -820,13 +822,38 @@ func (task *TaskRun) IsIntermittentExitCode(c int64) bool {
 }
 
 func (task *TaskRun) ExecuteCommand(index int) *CommandExecutionError {
+	c := task.Commands[index]
 	task.Infof("Executing command %v: %v", index, task.formatCommand(index))
-	log.Print("Executing command " + strconv.Itoa(index) + ": " + task.Commands[index].String())
+	log.Print("Executing command " + strconv.Itoa(index) + ": " + c.String())
 	cee := task.prepareCommand(index)
 	if cee != nil {
 		panic(cee)
 	}
-	task.result = task.Commands[index].Execute()
+	go c.MonitorResources(func(previouslyWarned bool) bool {
+		if config.DisableOOMProtection {
+			if !previouslyWarned {
+				task.Warn("Sustained memory usage above 90%!")
+				task.Warn("OOM protections are disabled, continuing task...")
+			}
+			return false
+		} else {
+			task.Warn("Sustained memory usage above 90%!")
+			task.Warn("Aborting task to prevent OOM issues...")
+		}
+		err := task.StatusManager.Abort(
+			&CommandExecutionError{
+				Cause:      errors.New("task aborted due to sustained memory usage above 90%"),
+				Reason:     internalError,
+				TaskStatus: aborted,
```

Comment:
Does `aborted` cause a rerun to happen automatically? (I presume so...just double checking!)

Reply:
I don't think we should conflate this with workerShutdown, we don't necessarily want to auto-retry on OOM IMO.

Reply:
That's also a good point Julien.

Reply:
Agree, `failed` is probably the correct task status.

Reply:
TaskStatus should be `failed` (so task resolves as failed, not as exception).